pkg/github/github.go

```diff
@@ -127,13 +127,31 @@ type Client interface {
 	) (*github.IssueComment, *github.Response, error)
 }
 
+// TODO: we should clean up the functions listed below and agree on the same
+// return type (with or without error):
+// - New
+// - NewWithToken
+// - NewEnterprise
+// - NewEnterpriseWithToken
```

Comment:
I put this as `TODO` for backwards compatibility reasons. Are we okay with breaking API? I know that at least [zeitgeist](https://github.com/kubernetes-sigs/zeitgeist) imports `pkg/github`.